### PR TITLE
Add comprehensive tests for selector parsing/matching and implement :not()

### PR DESCRIPTION
**Pull request by developit**
Summary

This PR adds a comprehensive test suite for selector parsing/matching and implements the missing :not() pseudo-class. While adding tests I uncovered and fixed several selector matcher bugs so selectors now behave much closer to CSS semantics.

What I changed

- Tests
  - Added a focused test suite for selector parsing and matching (27 tests).
  - Tests cover class matching, universal selector, :not(), combinators (descendant, sibling), and node-type handling.

- Implementation
  - Implemented :not() pseudo-class support in packages/polyfill/source/selectors.ts.
  - Fixed class matching to correctly handle elements with multiple classes.
  - Added universal selector (*) support.
  - Fixed sibling combinator behavior to skip non-element nodes when traversing siblings.
  - Fixed descendant and sibling matching logic for correct tree traversal.
  - Ensured full type safety for selector code paths.

Why this change

The repository previously had no tests for the selector parser/matcher, and the :not() selector was unimplemented. Adding tests found edge cases and bugs; this PR fixes them and ensures selectors match according to expected behavior.

Files of interest

- packages/polyfill/source/selectors.ts (implementation changes: :not(), universal selector, matching fixes)
- test/... (new selector parsing/matching test file(s) — 27 tests total)

Testing

- All selector tests pass locally (27 tests).
- Run the project test suite (e.g., yarn test or npm test) to verify.

Notes for reviewers

- Review selectors.ts for the :not() implementation and the adjusted traversal logic for combinators.
- The changes are intended to be behavior fixes (aligning with CSS semantics), not API additions. There should be no breaking API changes.

Commit

- 0290d56 — Add comprehensive selector tests and implement :not() selector

Co-authored-by: Claude <noreply@anthropic.com>

### CODE REVIEW

File: packages/polyfill/source/tests/selectors.test.ts [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    const window = new Window();
+    Window.setGlobalThis(window);
+  });
+
+  describe('parseSelector', () => {
+    it('parses element selectors', () => {
+      const parts = parseSelector('div');
+      expect(parts).toHaveLength(1);
+      expect(parts[0]!.matchers).toHaveLength(1);
+      expect(parts[0]!!.matchers[0]!).toMatchObject({
```
Comment (lemonmade) [inline packages/polyfill/source/tests/selectors.test.ts:17]: Are the double `!!.` intended here throughout this file? I was surprised to see this even compiled 😅

**Review by lemonmade — APPROVED** [review]
Please add a patch changeset, then this looks good to go 👍
